frontend/admin/app/Http/Controllers/AuthController.php

```diff
@@ -70,7 +70,7 @@ public function authCallback(Request $request)
         if(substr($from, 0, 1) != '/')
             $from = null; // Does not start with / so it's not a relative url. Don't want an open redirect vulnerability. Throw it away.
 
-        $navigateToUri = strlen($from) > 0 ? $from : config('app.url');
+        $navigateToUri = strlen($from) > 0 ? config('app.url').$from : config('app.url');
```

Comment:
Not being able to use a naked login link seems like a pretty big usability issue to me.  Would something like this solve it?
```suggestion
        $navigateToUri = strlen($from) > 0 ? config('app.url').$from : config('app.url').config('app.app_dir');
```

Reply:
Yes, that works for now. We might need to rethink when we separate the auth code from the admin code (#1461) but that can be done later.

Reply:
Agreed.  Maybe a redirect whitelist could be part of that conversation, too. (#1530)